pulsar-functions/java-examples/src/main/java/org/apache/pulsar/functions/api/examples/CounterFunction.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.functions.api.examples;
+
+import org.apache.pulsar.functions.api.Context;
+import org.apache.pulsar.functions.api.PulsarFunction;
+
+public class CounterFunction implements PulsarFunction<String, Void> {
+    @Override
+    public Void process(String input, Context context) throws Exception {
```

Comment:
Would it make sense to add a specialization of `PulsarFunction` interface for a void function? 

In that way, we won't have the `Void` return type with the meaningless `return null` .

Reply:
That is a good question. I think it falls into a question, do we support java.util.Consumer<T>, which is a void native function.

/cc @srkukarni

Reply:
Yes `java.util.Consumer` is fine and probably easy to support (without SDK), though it would need a similar void correspective of `PulsarFunction`, like `PulsarConsumer`...  which might get confusing.. :)